### PR TITLE
Split ChartType components into files

### PR DESCRIPTION
**Pull request by oisincoveney**
This PR:
- Splits `ChartType` components into separate files
- Moves the logic of displaying chart types away from the sidebar component into its own component (for use within the SDK)
- Removes the dependence of ML elements from the inner components (i.e. `question`, `result`, `query`) so we can give more customization options to users in the future. For example, maybe they want to use a list of visualizations rather than the current grid of circles

I'll fix the tests tomorrow (PDT) but if y'all have feedback on the code I would appreciate it.

### CODE REVIEW

**Review by alxnddr — COMMENTED** [review]
Hey @oisincoveney 👋 The component separation looks good!

I found that it is not possible to change viz type: 
<video src="https://github.com/user-attachments/assets/bba8f4c7-ce4e-4848-bced-62417ae6110d" />

**Comment by oisincoveney** [comment]
> Hey @oisincoveney 👋 The component separation looks good!
> 
> I found that it is not possible to change viz type:
> 
>  Screen.Recording.2024-09-06.at.4.26.58.PM.mov

Yep, just ran into that one right now 😅 working on it right now!

**Comment by oisincoveney** [comment]
@alxnddr problem is solved! It was just a matter of some logic that shouldn't have been inside an `if` statement 🙃

File: frontend/src/metabase/query_builder/components/chart-type-selector/use-chart-type-visualizations.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  const visualization = visualizations.get(vizType);
+  return (
+    (result?.data &&
+      visualization?.isSensible?.(sanatizeResultData(result.data))) ||
+    false
+  );
+};
+
+export type GetSensibleVisualizationsProps = Pick<
+  IsSensibleVisualizationProps,
```
Comment (alxnddr) [inline frontend/src/metabase/query_builder/components/chart-type-selector/use-chart-type-visualizations.ts:75]: nit: I found it a bit harder to follow prop types definitions when they refer to each other like this using `Pick`. It seems that 1-2 props could've been defined in their own own interface without referring to other props

**Review by alxnddr — APPROVED** [review]
I left one minor comment, besides that looks good

**Pull request merged**